pkg/smokescreen/config.go

```diff
@@ -43,6 +43,8 @@ type Config struct {
 	StatsSocketFileMode          os.FileMode
 	StatsServer                  interface{} // StatsServer
 	ConnTracker                  *sync.Map   // The zero Map is empty and ready to use
+	IdleThresholdSec             time.Duration // Consider a connection idle if it has been inactive (no bytes transferred) for this many seconds.
+	WaitForAllIdleSec            time.Duration // Wait this many seconds before checking (again) to ensure all connections are idle before exiting.
```

Comment:
I think you won't need `WaitForAllIdleSec` if you implement one of https://github.com/stripe/smokescreen/pull/46#issuecomment-479958041